### ISSUE TITLE
Add content type dependency

### ISSUE THREAD

**Issue by sebastienros** [OrchardCMS/OrchardCore#1850]
Invalidates any time a content item for this content type has changed, been deleted or added.
To invalidate cached queries if a its results might be impacted by a change on a content item of a specific type.

/cc @paynecrl97

**Comment by paynecrl97** [comment]
Yep. It'd be good to also be able to specify which events should invalidated the cache. e.g. `content-type:blogpost.created`, `content-type:blogpost.edited`, or just `content-type:blogpost.unpublished` for every event.

e.g. you have a most recent blog post query, you only really need to invalidate this if a new post is published. Posts being removed or edited would be handled by the individual cache blocks around each summary display.
